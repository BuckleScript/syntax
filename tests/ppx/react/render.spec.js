runPrinter(__dirname, "jsx", true)
