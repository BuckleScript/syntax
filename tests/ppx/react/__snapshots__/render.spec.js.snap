// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aria.res 1`] = `
"let foo = (@warning(\\"-3\\") React.jsx)(
  ReactDOM.stringToComponent(\\"div\\"),
  {
    let ariaCurrent = #page
    {
      \\"aria-current\\": ariaCurrent: [
        | #page
        | #step
        | #location
        | #date
        | #time
        | #\\"true\\"
        | #\\"false\\"
      ],
    }
  },
)
"
`;

exports[`commentAtTop.res 1`] = `
"@obj external makeProps: (~msg: 'msg, unit) => {\\"msg\\": 'msg} = \\"\\" // test React JSX file

let make =
  (@warning(\\"-16\\") ~msg) => {
    (@warning(\\"-3\\") React.jsx)(
      {
        ReactDOM.stringToComponent(\\"div\\")
      },
      {
        let children = {msg->React.string}
        {\\"children\\": children: React.element}
      },
    )
  }
let make = {
  let \\\\\\"CommentAtTop\\" = (\\\\\\"Props\\": {\\"msg\\": 'msg}) => make(~msg=\\\\\\"Props\\"[\\"msg\\"])
  \\\\\\"CommentAtTop\\"
}
"
`;

exports[`data.res 1`] = `
"let foo = (@warning(\\"-3\\") React.jsx)(
  ReactDOM.stringToComponent(\\"div\\"),
  {
    let \\\\\\"data-foo\\" = \\"payload\\"
    {\\"data-foo\\": \\\\\\"data-foo\\": string}
  },
)
"
`;

exports[`error_badComponent.res 1`] = `
"=====Result==========================================

=====Errors=============================================
File \\"Error_badComponent\\", line 1:
Error: React: react.component calls can only decorate function definitions or component wrappers (forwardRef, memo).

========================================================"
`;

exports[`error_tooManyReactComponent.res 1`] = `
"=====Result==========================================

=====Errors=============================================
File \\"/Users/rickyvetter/code/syntax/tests/ppx/react/error_tooManyReactComponent.res\\", line 1, characters 0-105:
Error: React: only one react.component call can exist on a component at one time

========================================================"
`;

exports[`externalWithCustomName.res 1`] = `
"module Foo = {
  @obj
  external componentProps: (
    ~a: int,
    ~b: string,
    unit,
  ) => {\\"a\\": int, \\"b\\": string} = \\"\\"
  @module(\\"Foo\\")
  external component: React.componentLike<
    {\\"a\\": int, \\"b\\": string},
    React.element,
  > = \\"component\\"
}

let t = (@warning(\\"-3\\") React.jsx)(
  Foo.component,
  Foo.componentProps(~a=1, ~b={\\"1\\"}, ()),
)
"
`;

exports[`fragment.res 1`] = `
"let foo =
  @warning(\\"-3\\")
  React.jsxs(
    React.Fragment.make,
    React.Fragment.makeProps(
      ~children=React.array([
        (@warning(\\"-3\\") React.jsx)(ReactDOM.stringToComponent(\\"div\\"), ()),
        (@warning(\\"-3\\") React.jsx)(ReactDOM.stringToComponent(\\"span\\"), ()),
      ]),
      (),
    ),
  )
"
`;

exports[`innerModule.res 1`] = `
"module Bar = {
  @obj external makeProps: (~a: 'a, ~b: 'b, unit) => {\\"a\\": 'a, \\"b\\": 'b} = \\"\\"
  let make =
    (@warning(\\"-16\\") ~a, @warning(\\"-16\\") ~b, _) => {
      Js.log(\\"This function should be named \`InnerModule.react$Bar\`\\")
      (@warning(\\"-3\\") React.jsx)(ReactDOM.stringToComponent(\\"div\\"), ())
    }
  let make = {
    let \\\\\\"InnerModule$Bar\\" = (\\\\\\"Props\\": {\\"a\\": 'a, \\"b\\": 'b}) =>
      make(~b=\\\\\\"Props\\"[\\"b\\"], ~a=\\\\\\"Props\\"[\\"a\\"], ())
    \\\\\\"InnerModule$Bar\\"
  }
  @obj
  external componentProps: (~a: 'a, ~b: 'b, unit) => {\\"a\\": 'a, \\"b\\": 'b} = \\"\\"

  let component =
    (@warning(\\"-16\\") ~a, @warning(\\"-16\\") ~b, _) => {
      Js.log(\\"This function should be named \`InnerModule.react$Bar$component\`\\")
      (@warning(\\"-3\\") React.jsx)(ReactDOM.stringToComponent(\\"div\\"), ())
    }
  let component = {
    let \\\\\\"InnerModule$Bar$component\\" = (\\\\\\"Props\\": {\\"a\\": 'a, \\"b\\": 'b}) =>
      component(~b=\\\\\\"Props\\"[\\"b\\"], ~a=\\\\\\"Props\\"[\\"a\\"], ())
    \\\\\\"InnerModule$Bar$component\\"
  }
}
"
`;

exports[`topLevel.res 1`] = `
"@obj external makeProps: (~a: 'a, ~b: 'b, unit) => {\\"a\\": 'a, \\"b\\": 'b} = \\"\\"
let make =
  (@warning(\\"-16\\") ~a, @warning(\\"-16\\") ~b, _) => {
    Js.log(\\"This function should be named 'TopLevel.react'\\")
    (@warning(\\"-3\\") React.jsx)(ReactDOM.stringToComponent(\\"div\\"), ())
  }
let make = {
  let \\\\\\"TopLevel\\" = (\\\\\\"Props\\": {\\"a\\": 'a, \\"b\\": 'b}) =>
    make(~b=\\\\\\"Props\\"[\\"b\\"], ~a=\\\\\\"Props\\"[\\"a\\"], ())
  \\\\\\"TopLevel\\"
}
"
`;
